docs/code-conventions/sql.md

```diff
@@ -0,0 +1,178 @@
+
+!!! info "Name length"
+
+    PostgreSQL's maximum name length is 63 characters, which should be sufficient for most elements, even foreign key constraints with multiple columns.
+
+    In case a name constructed according to the rules above ends up being too long, abbreviate or truncate some segments.
+    However, double check that it is still obvious which column each name segment refers to, and that there is no ambiguity.
+
+### Sequences and triggers
+
+- Sequence names consist of the corresponding table name and the suffix `__seq` (same as ed-DB-119 but with `__` instead of `_`).
+- Trigger names consist of the table name, two underscores and a short but meaningful title.
+  We intentionally do not add prefixes (like `trg_` from ed-DB-119) or suffixes (e.g. `bu` for "before update" from ed-DB-119). 
+
+    !!! example "Examples"
+        - `book__insert`, `book__update` and `book__delete`
+        - `my_cool_table__track_history`
+        - `payment__trigger_processing`
+
+
+## Table and column declarations
+
+- Define primary keys, foreign key constraints, check constraints and indexes (unique or otherwise) explicitly, not as part of a column definition.
+- Always specify the name of keys, constraints and indexes.
+  Letting the RDBMS generate names would prevent developers from documenting their intention and often makes code hard to comprehend. 
+- Foreign key constraints
+    - Each relation between tables must be covered by a foreign key constraint (see ed-DB-107).
+    - Only specify the clauses that deviate from the default, e.g. include `on delete cascade` but don't add `on update no action`.
+    - If the referenced column is the primary key, specify only the table name without a column name:
+      ```sql
+      constraint book__fk__author_id
+          foreign key (author_id)
+              references author   -- leave out `(id)` here
+              on delete cascade
+      ```
+- Primary keys should always be a [surrogate key](https://en.wikipedia.org/wiki/Surrogate_key), also known as a synthetic key (see ed-DB-108).
+    - Using [natural/business keys](https://en.wikipedia.org/wiki/Natural_key) (like email addresses, or an employee ID generated by another system) may feel more intuitive at first, but it often causes problems later on.
+    - Instead, include the natural/business key as a data column, possibly with an index.
+      However, if the data originates from another system, be careful with unique indexes as you may not be able to trust that system to enforce uniqueness at all times. 
+    - This rule applies even if (or _especially if_) you cannot think of a scenario where the "obvious" natural/business key value of a row would change.
+- Columns that hold ID/key values generated by another system should always unambiguously identify that system.
+    - You may feel that it's obvious to everybody that `external_id` or `source_id` refers to the company's "SplineReticulator Directory" product.
```

Comment:
We already have an example of where we literally _have_ a `source_id`, on purpose. However, the table in question also has a `source_system` column, as it's meant to reference data in different system, one per row.

I'm not sure whether we need to add an exclusion to the general rule, just for that particular case. The intention could be hard to explain in a general document like this.

I'll leave it to you, whether you want to somehow include this.